src/Game.tsx

```diff
@@ -1,16 +1,33 @@
 import React from 'react';
 import { Board } from './Board';
-import { useGameState } from './GameContext';
 
-type Props = {};
+type State = {
+  board: string[],
+  xToPlay: boolean
+};
 
-export const Game = (props: Props): JSX.Element => {
-  const state = useGameState();
+export class Game extends React.Component<{}, State> {
+  public constructor(props: {}) {
```

Comment:
When I was reading the React docs, their introduction of Hooks makes it sound like that FunctionalComponents are the future and classes are a "legacy" approach. That's why I was trying to stay away from them. https://reactjs.org/docs/hooks-intro.html#no-breaking-changes

Turning the chair around though; I can see how it'd be useful to be familiar with both ways of doing when joining existing projects/codebases. This ecosystem moves so fast that FC's will probably be out of style by next year anyway in favor of something else. 😇

Reply:
@beporter Yeah, we work with legacy codebases, so hooks aren't in use with them. It's good to learn about context/hooks, but their applicability in our code-bases is limited. That's why I was pushing you _not_ to. And they're new in 16.8, which I don't think we're using for most of our code bases....

And tbh, with react, I'd like to wait a few version before using those bits in prod.

(also, full disclosure, I had no idea about hooks into you brought it up)